src/app/pages/PerpetualPage/components/ClosePositionDialog/components/TradeFormStep.tsx

```diff
@@ -48,7 +48,18 @@ export const TradeFormStep: TransitionStep<ClosePositionDialogStep> = ({
   const dispatch = useDispatch();
   const { t } = useTranslation();
 
-  const currentPairId = usePerpetual_getCurrentPairId();
+  const { useMetaTransactions } = useSelector(selectPerpetualPage);
+
+  const inMaintenance = usePerpetual_isTradingInMaintenance();
+
+  const { changedTrade, trade, onChange } = useContext(
+    ClosePositionDialogContext,
+  );
+
+  const tradePair = PerpetualPairDictionary.get(
+    trade?.pairType || PerpetualPairType.BTCUSD,
+  );
+  const currentPairId = tradePair.id;
```

Comment:
This is unnecessary, delete it pls. You already have pair retrieval (line 75), just change it a little (`currentPairType` should be a fallback) and move perpetual data retrieval below the `pair` (use `pair.id` instead of `currentPairId`).